app/src/main/java/com/example/jasmiensofiecels/wordly/view/di/di/component/ApplicationComponent.java

```diff
@@ -0,0 +1,35 @@
+package com.example.jasmiensofiecels.wordly.view.di.di.component;
+
+import android.app.Application;
+
+import com.example.jasmiensofiecels.wordly.WordlyApplication;
+import com.example.jasmiensofiecels.wordly.view.di.di.module.ActivityBindingModule;
+import com.example.jasmiensofiecels.wordly.view.di.di.module.AppModule;
+
+import dagger.BindsInstance;
+import dagger.Component;
+import dagger.android.AndroidInjectionModule;
+import dagger.android.AndroidInjector;
+
+/**
+ * The purpose of this interface is to represent the injector class.
+ * It will build the component graph, which will provided injected instances through modules.
+ *
+ * Created by Jasmien Cels on 12/03/2018.
+ */
+
```

Comment:
I didn't care about this on the AND app but pulling no punches out here in the real world.

This should be scoped with `@Singleton`. We don't want more than one instance of `AppComponent` being created as we only want one of these per app. Imagine it created another `AppModule` class? All that memory

Upon changing this to `@Singleton` you'll see that it will throw an error saying that scoped dependencies cannot depend on unscoped. You'll need to create custom scopes for each Activity. Scopes names can be anything, but by declaring `@Singleton` on top, you then place an annotation on a child, usually belonging to an `Activity` unless you want it to be a singleton (like a service) where you'll scope it with something else.

I feel like I've rambled. Hope this makes sense.